crypto/evp/m_sigver.c

```diff
@@ -110,23 +94,79 @@ static int do_sigver_init(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
     if (supported_sig == NULL)
         supported_sig = locpctx->keytype;
 
+    if (!evp_keymgmt_can_export(tmp_keymgmt)) {
+        /*
+         * The KEYMGMT cannot export, therefore we must only fetch an
+         * implementation from the same provider.
+         */
+        tmppropq
+            = ossl_property_str_for_provider(locpctx->propquery,
+                                             EVP_KEYMGMT_get0_provider(tmp_keymgmt));
+        if (tmppropq == NULL) {
+            ERR_raise(ERR_LIB_EVP, ERR_R_INTERNAL_ERROR);
+            goto err;
+        }
+    }
```

Comment:
This has me deeply worried, 'cause it essentially means that if the calling application has asked that the *operation* should happen with a signature method matching properties that *it* asks for, we now allow ourselves to turn around and essentially say "we know better than you what you want!" instead of simply saying back that "this combination simply doesn't work".  That's a behavior change that I'd call surprising.

Reply:
Err, no. We are *adding* properties to the fetch here, not removing any. We will still only return an algorithm that would have been eligible to be returned anyway. The additional property will just further restrict what is allowed to be returned.